confettiView.js

```diff
@@ -41,27 +42,36 @@ class ConfettiView extends Component {
   }
 
   removeConfetti(key) {
-       let {confettis} = this.state;
+       let {confettis, onComplete} = this.state;
        let {confettiCount} = this.props;
        let index = confettis.findIndex(confetti => {return confetti.key === key});
        confettis.splice(index, 1);
        this.setState({confettis});
        if(key === confettiCount - 1) {
          this.confettiIndex = 0;
        }
+       if(confettis.length === 0 && onComplete && typeof onComplete === 'function') {
+         onComplete();        
+       }
   }
 
   stopConfetti ()
   {
     this.shouldStop = true;
+    this.confettiIndex = 0;
+    const { onComplete } = this.state;
+    if(onComplete && typeof onComplete === 'function') {
+      onComplete();        
+    }
+    this.setState({ confettis: [], onComplete: null });
   }
```

Comment:
Immediately stopping the confetti so that it can be restarted immediately via `startConfetti`